src/main/java/graphql/schema/DefaultGraphqlTypeComparatorRegistry.java

```diff
@@ -18,9 +19,45 @@
 @PublicApi
 public class DefaultGraphqlTypeComparatorRegistry implements GraphqlTypeComparatorRegistry {
 
-    public static final Comparator<GraphQLSchemaElement> DEFAULT_COMPARATOR;
-    static {
-        DEFAULT_COMPARATOR = Comparator.comparing(element -> {
+    // This sensible order was taken from the original SchemaPrinter code.  It ordered the types in this manner
+    private static final ImmutableMap<Class<? extends GraphQLSchemaElement>, Integer> SENSIBLE_ORDER =
+            ImmutableMap.<Class<? extends GraphQLSchemaElement>, Integer>builder()
+                    .put(GraphQLDirective.class, 1)
```

Comment:
This doesn't specify unfortunately how "schema { query: Foo ..}" element will be printed.